keras_cv/models/backbone.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for Backbone models."""
+
+import os
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv.utils.python_utils import classproperty
+from keras_cv.utils.python_utils import format_docstring
+
+
+class Backbone(keras.Model):
+    """Base class for Backbone models."""
```

Comment:
perhaps a little description of what we mean by "backbone" would be good - especially for those who stumble on it by browsing the code base.